drivers/adc/adc_smartbond_gpadc.c

```diff
@@ -117,9 +117,7 @@ static inline void gpadc_smartbond_pm_policy_state_lock_get(const struct device
 {
 
 #if defined(CONFIG_PM_DEVICE)
-#if defined(CONFIG_PM_DEVICE_RUNTIME)
 	pm_device_runtime_get(dev);
-#endif
 
 	if (!atomic_test_and_set_bit(data->pm_policy_state_flag, 0)) {
```

Comment:
It seems that unbalanced calls should not occur and so atomic operations are not really needed. If so, `pm_device_runtime_get/put` should be encapsulated inside the atomic operations. 

```
static int adc_smartbond_read()
{
	gpadc_smartbond_pm_policy_state_lock_get(dev, data);
	adc_context_lock(&data->ctx, false, NULL);
        ...
}
```
PM constraints should take place after context lock and not before so there are no chances for unbalanced calls. The same goes for `adc_smartbond_read_async`.

Reply:
Removed atomic operation.